type-mappings/src/macros/type_macros.rs

```diff
@@ -0,0 +1,326 @@
+#[macro_export]
+macro_rules! generate_algebraic_types {
+    ($curve: ident, $curve_parameters: ty) => {
+
+        // Basic algebraic types
+        pub type FieldElement = <$curve_parameters as ModelParameters>::BaseField;
+
+        pub type ScalarFieldElement = <$curve_parameters as ModelParameters>::ScalarField;
+
+        pub type Group = $curve;
+        pub type GroupProjective = <$curve as AffineCurve>::Projective;
+
+        pub type BigInteger = <<FieldElement as Field>::BasePrimeField as PrimeField>::BigInt;
+
+        // Basic constants
+        pub const FIELD_CAPACITY: usize = <FieldElement as PrimeField>::Params::CAPACITY as usize;
+        pub const FIELD_SIZE: usize = ((<FieldElement as PrimeField>::Params::MODULUS_BITS
+            + <FieldElement as PrimeField>::Params::REPR_SHAVE_BITS)
+            as usize)
+            / 8;
+        pub const SCALAR_FIELD_SIZE: usize =
+            ((<ScalarFieldElement as PrimeField>::Params::MODULUS_BITS
+                + <ScalarFieldElement as PrimeField>::Params::REPR_SHAVE_BITS)
+                / 8) as usize;
+        pub const GROUP_SIZE: usize = 2 * FIELD_SIZE + 1;
+        // ceil ((MODULUS_BITS + 2 (1 bit for infinity and 1 for odd/even y))/8)
+        pub const GROUP_COMPRESSED_SIZE: usize = FIELD_SIZE
```

Comment:
No: serializing a compressed point requires to serialize one of the two coordinates (whose size is FIELD_SIZE) and at most (for the current curve types we have) two boolean flags (infinity and other coordinate sign). Since REPR_SHAVE_BITS are padding bits and are not used anyway, the serialize_compressed will re-use those bits by storing the flags inside them if they are enough (e.g. =2). If they are less than 2, than the code will be forced to serialize the flags using an additional byte

Reply:
Maybe the comment is misleading, I will change that